kalite/coachreports/static/js/coachreports/student_progress.js

```diff
@@ -101,10 +102,24 @@ var StudentProgressContainerView = Backbone.View.extend({
 
     initialize: function() {
         this.listenTo(this.collection, 'add', this.add_one);
+        this.count=0;
+        var self=this;
 
         this.render();
 
-        this.collection.fetch();
+        this.collection.fetch({
+                success: function() {
+                   
+                    self.count=self.collection.length;    //number of reports of quizes/videos
+                
+                     if(self.count==0)                    //if the student visits the my progress page before attempting any quizes/videos
```

Comment:
You could just test `self.collection.length` directly, no need to create a new variable.